src/test/unit_tests/test_agent_blocks_verifier.js

```diff
@@ -13,35 +15,51 @@ const AgentBlocksVerifier = require('../../server/bg_services/agent_blocks_verif
 const mongo_utils = require('../../util/mongo_utils');
 const mongodb = require('mongodb');
 const config = require('../../../config');
+const { ChunkDB, BlockDB } = require('../../server/object_services/map_db_types');
 
 class VerifierMock extends AgentBlocksVerifier {
-
-    constructor(blocks = [], nodes = [], chunks = [], chunk_coder_configs = [], test_suffix = '') {
+    /**
+     * 
+     * @param {nb.BlockSchemaDB[]} blocks 
+     * @param {nb.NodeAPI[]} nodes 
+     * @param {nb.ChunkSchemaDB[]} chunks 
+     * @param {nb.Pool[]} pools 
+     * @param {string} test_suffix 
+     */
+    constructor(blocks = [], nodes = [], chunks = [], pools = [], test_suffix = '') {
         super(`VerifierMock-${test_suffix}`);
-        console.log('VerifierMock INIT', test_suffix, blocks, nodes, chunks, chunk_coder_configs);
+        console.log('VerifierMock INIT', test_suffix, blocks, nodes, chunks);
         this.blocks = blocks;
         this.nodes = nodes;
         this.chunks = chunks;
-        this.chunk_coder_configs = chunk_coder_configs;
+        this.pools = pools;
+        /** @type {nb.BlockSchemaDB[]} */
         this.verified_blocks = [];
     }
 
-    iterate_all_blocks(marker, limit, deleted_only) {
+    /**
+     * 
+     * @param {nb.ID} marker 
+     * @param {number} limit 
+     * @param {boolean} deleted_only 
+     * @returns {Promise<nb.BlockSchemaDB[]>}
+     */
+    async iterate_all_blocks(marker, limit, deleted_only) {
         assert(marker === undefined || marker === null || this.is_valid_md_id(marker),
             'Marker not null or valid ObjectId');
         assert(limit === config.AGENT_BLOCKS_VERIFIER_BATCH_SIZE, 'Wrong verifier config limit');
         assert(deleted_only === false, 'Verifier should check only non deleted and non reclaimed blocks');
         // TODO: Pay attention to marker and limit
-        return P.resolve(this.blocks.filter(block => (deleted_only ? block.deleted : !block.deleted)));
+        return this.blocks.filter(block => (deleted_only ? block.deleted : !block.deleted));
     }
 
     is_valid_md_id(id_str) {
         return mongodb.ObjectId.isValid(id_str);
     }
 
-    get_by_id(id_str) {
-        return this.chunk_coder_configs.find(coder => String(coder._id) === String(id_str));
-    }
+    // get_by_id(id_str) {
+    //     return this.chunk_coder_configs.find(coder => String(coder._id) === String(id_str));
```

Comment:
Remove this comment

Reply:
Notice that we used this in order to override the population of chunk coder config.
Now since you've removed this from the bg worker and you are doing that inside the population you should check that it is working and is getting populated since I couldn't find the code which handles that.

Reply:
no need anymore. used ccc for block_md digest_type. now we mock just it.